src/protocol.h

```diff
@@ -230,6 +230,12 @@ extern const char *GETBLOCKTXN;
  * @since protocol version 70014 as described by BIP 152
  */
 extern const char *BLOCKTXN;
+/**
+ * Indicates that a node prefers to relay transactions via wtxid, rather than
+ * txid.
```

Comment:
Yes, you made a good observation -- our fetching of parents of orphan transactions must be by txid, rather than wtxid.  I did try to carefully word the BIP to avoid referencing this situation:

> After a node has sent and received a "wtxidrelay" message to/from a given peer, the node is required to use the MSG_WTX inv-type when announcing transactions to that peer, or requesting announced transactions from that peer.

So the BIP says that newly announced transactions must be by wtxid.  And if your peer announces a transaction by wtxid, you must fetch it by wtxid (and not txid).  The BIP is silent on requests for transactions that are not announced -- we could make this explicit but this has never been documented behavior, and I'm not sure we should commit to a certain handling of this kind of thing anyway...  But open to suggestions.

Reply:
I would amend the BIP to make its scope explicitly restrained. Add to 5) "Note: this rule doesn't cover cases where a peer needs to request transaction for validation purposes but don't know its wtxid yet (e.g missing parents of a wtxid-announced child)" ? At least to avoid someone raising same concern in the future and saying we don't commit on anything wrt to this issue.